test/jdk/java/awt/Window/WindowIconUpdateOnDPIChanging/WindowIconUpdateOnDPIChangingTest.java

```diff
@@ -0,0 +1,147 @@
+            + "      with a 150% DPI scaling, or \"32\" on a screen with a 200% DPI scaling."
+            + "      <br>If so, press <b>PASS</b>, otherwise press <b>FAIL</b>.</li>"
+            + "<ol>\n"
+            + "</body>\n"
+            + "</html>\n";
+
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeLater(WindowIconUpdateOnDPIChangingTest::createUI);
+        if (!countDownLatch.await(15, TimeUnit.MINUTES)) {
+            frame.dispose();
+            throw new RuntimeException("Timed out!");
+        } else if (!testResult) {
+            throw new RuntimeException("Test fails!");
+        }
+    }
+
+    private static void createUI() {
+        frame = new JFrame("Window Icon Update Test");
+        frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+        frame.setSize(640, 480);
+        frame.addWindowListener(new WindowAdapter() {
+            public void windowClosed(WindowEvent e) {
+                countDownLatch.countDown();
+            }
+        });
+        frame.getContentPane().setLayout(new BorderLayout());
+        frame.getContentPane().add(createInstrumentsPane(), BorderLayout.CENTER);
+        frame.getContentPane().add(createControlPanel(), BorderLayout.SOUTH);
+        frame.setIconImages(IntStream.rangeClosed(16, 32).mapToObj(size -> createIcon(size)).toList());
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static JTextPane createInstrumentsPane() {
+        JTextPane instructionsPane = new JTextPane();
+        instructionsPane.setContentType("text/html");
+        instructionsPane.setText(INSTRUCTIONS);
+        instructionsPane.setEditable(false);
+        instructionsPane.setBorder(new EmptyBorder(10, 10, 10, 10));
+        return instructionsPane;
+    }
+
+    private static JPanel createControlPanel() {
+        JButton passButton = new JButton("Pass");
+        passButton.addActionListener(e -> testResult = true);
+        passButton.addActionListener(e -> frame.dispose());
+
+        JButton failButton = new JButton("Fail");
+        failButton.addActionListener(e -> frame.dispose());
+
+        JPanel controlPanel = new JPanel(new FlowLayout(FlowLayout.CENTER));
+        controlPanel.add(passButton);
+        controlPanel.add(failButton);
+
+        return controlPanel;
+    }
+
+    /**
+     * Creates an icon of the size specified. The size is drawn at the center of the icon.
+     *
+     * @param size the size of the icon
+     */
+    private static Image createIcon(int size) {
+        BufferedImage image = new BufferedImage(size, size, BufferedImage.TYPE_INT_ARGB);
+        Graphics2D g = image.createGraphics();
+        g.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
+        g.setColor(Color.WHITE);
+        g.fillRect(0, 0, size, size);
+        g.setFont(new Font("dialog", Font.BOLD, 12));
```

Comment:
Could you please also start the name of the logical font with the capital letter?
```suggestion
        g.setFont(new Font("Dialog", Font.BOLD, 12));
```